### ISSUE TITLE
External etcd preflight check does not surface errors

### ISSUE THREAD

**Issue by alexbrand** [kubernetes/kubeadm#719]
## What keywords did you search in kubeadm issues before filing this one?

etcd preflight

## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`): `v1.9.3`

**Environment**:
- **Kubernetes version** (use `kubectl version`): `v1.9.3`
- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04.3 LTS
- **Kernel** (e.g. `uname -a`): `4.4.0-1049-aws`

## What happened?

The external etcd preflight check failed with a confusing error I misconfigured the etcd endpoints in the kubeadm config file (used `http://etcd-node` instead of `http://etcd-node:2379`). 

The error was not helpful nor indicative of what the actual issue was. Furthermore, it seems like kubeadm assumed it was able to connect to the etcd cluster and attempted to parse the version:

```
root@ip-172-31-15-120:~# kubeadm init --config kubeadm-config.yaml
[init] Using Kubernetes version: v1.9.3
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks.
	[WARNING FileExisting-crictl]: crictl not found in system path
[preflight] Some fatal errors occurred:
	[ERROR ExternalEtcdVersion]: couldn't parse external etcd version "": Version string empty
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`

```

## What you expected to happen?
The error message to include the actual error that happened when trying to connect to etcd (e.g. connection refused, no route to host, etc).

## How to reproduce it (as minimally and precisely as possible)?

```
cat > kubeadm-config.yaml <<EOF
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
etcd:
  endpoints:
  - http://non-existent-host
EOF

kubeadm init --config kubeadm-config.yaml
```

**Comment by kamarajuPrathi** [comment]
@alexbrand I am facing the same issue, did you guys find a solution for it ?

**Comment by alexbrand** [comment]
@kamarajuPrathi There is an open PR with a fix here: https://github.com/kubernetes/kubernetes/pull/60585

**Comment by Klmanjaro** [comment]
@timothysc: is the fix for this issue available in kubeadm v1.11 ? When is that going to be available? 
thanks

**Comment by Klmanjaro** [comment]
Never mind. I have figured out the issue. typo :-(
